qiskit/extensions/hamiltonian_gate.py

```diff
@@ -113,8 +113,8 @@ def _define(self):
         self.definition = qc
 
     def qasm(self):
-        """Raise an error, as QASM is not defined for the HamiltonianGate."""
-        raise ExtensionError("HamiltonianGate has no QASM definition.")
+        """Raise an error, as OpenQASM2 is not defined for the HamiltonianGate."""
+        raise ExtensionError("HamiltonianGate has no OpenQASM2 definition.")
```

Comment:
"QASM [or OpenQASM 2] is not defined for the HamiltonianGate" is a bit clumsy. Maybe something like

"OpenQASM 2.0 does not support Hamiltonian gates".

But in fact it looks like typically (or always) all the parameters are explicit numbers so a unitary could be computed. And OpenQASM 2.0 could be emitted for this. So, rather than a limitation of OpenQASM, aren't we choosing not to implement it. In this case,  a more accurate error message might be

"Converting `HamiltonianGate` to OpenQASM 2.0 is not supported."

Also, in "Raise an error, as OpenQASM2 is not defined for the HamiltonianGate." shouldn't there be some kind of markup on HamiltonianGate?

Reply:
With the deprecation of `Instruction.qasm` #10399, I think this comment became obsolete.